src/viam/sdk/robot/service.cpp

```diff
@@ -147,31 +148,45 @@ ::grpc::Status RobotService_::StopAll(::grpc::ServerContext* context,
                                       ::viam::robot::v1::StopAllResponse* response) {
     const ResourceName r;
     std::unordered_map<std::string, AttributeMap> extra;
-    grpc::StatusCode status = grpc::StatusCode::OK;
     for (const auto& ex : request->extra()) {
         const google::protobuf::Struct& struct_ = ex.params();
         const AttributeMap value_map = struct_to_map(struct_);
         const std::string name = ex.name().SerializeAsString();
         extra.emplace(name, value_map);
     }
 
+    grpc::StatusCode status = grpc::StatusCode::OK;
+    std::string status_message;
+
     for (const auto& r : resource_manager()->resources()) {
         const std::shared_ptr<Resource> resource = r.second;
         const ResourceName rn = resource->get_resource_name(resource->name());
         const std::string rn_ = rn.SerializeAsString();
         if (extra.find(rn_) != extra.end()) {
-            grpc::StatusCode status = resource->stop(extra.at(rn_));
-            if (status != grpc::StatusCode::OK) {
-                // NOLINTNEXTLINE(clang-analyzer-deadcode.DeadStores)
-                status = resource->stop();
+            try {
+                resource->stop(extra.at(rn_));
+            } catch (const std::runtime_error& err) {
+                try {
+                    status_message = err.what();
+                    // NOLINTNEXTLINE(clang-analyzer-deadcode.DeadStores)
+                    resource->stop();
+                } catch (std::runtime_error& err) {
+                    status_message = err.what();
+                    status = grpc::UNKNOWN;
+                }
```

Comment:
The rationale as I recall was, "if it failed it could be because the `extra` was malformed in some way, so try calling the standard `stop` method as well just to be safe." This was from a while ago however; the "call `stop` twice" logic was in there before. My goal in this PR was to minimize changes to logic.